cli.js

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env node
+var fs = require('fs')
+var minimist = require('minimist')
+var fastq = require('./index')
+
+var argv = minimist(process.argv.slice(2), {
+  alias: {
+    help: 'h'
+  }
+})
+
+if (argv.help || argv._.length === 0) {
+  console.log(
+        'Usage: bionode-fastq <options> <fastq file [required]> <output file>\n' +
+        'If no output is provided, the result will be printed to stdout\n'
+    )
+  process.exit(0)
+}
+
+// if there is an output file present, write to the file. Otherwise write to stdout
+var output = argv._[1] ? fs.createWriteStream(argv._[1]) : process.stdout
+
+var fq = fastq.read(argv._[0])
+
+process.stdin.setEncoding('utf8')
+
+fq.on('data', function (data) {
```

Comment:
using `.on('data')` puts a stream into streams 1 mode - it is preferred to use a pipe instead. It would like something like

```javascript
const through = require('through2')

fq.pipe(through.obj(function (data, enc, next) {
  this.push(JSON.stringify(data) + '\n')
  next()
  // alternatively, next(null, JSON.stringify(data) + '\n')
})
  .pipe(writeStream)
```

There is also [ndjson.stringify](https://www.npmjs.com/package/ndjson#ndjsonserialize--ndjsonstringify) which I believe does the [same thing](https://github.com/maxogden/ndjson/blob/master/index.js#L27). (and it handles linux vs windows EOL)

So: use `ndjson` module.